qdrant-landing/content/articles/discovery-search.md

```diff
@@ -0,0 +1,93 @@
+
+Wait a second, what has just happened? These pictures have __nothing__ to do with burgers, and still, they appear on the first results. Is the demo broken?
+
+Turns out, multimodal encoders [might not work how you expect them to](https://twitter.com/metasemantic/status/1356406256802607112). Images and text are embedded in the same space, but they are not necessarily close to each other. This means that we can create a mental model of the distribution as two separate planes, one for images and one for text.
+
+![Mental model of CLIP embeddings](/articles_data/discovery-search/clip-mental-model.png)
+
+This is where discovery excels, because it allows us to constrain the space considering the same mode (images) while using a target from the other mode (text).
+
+![Cross-modal search with discovery](/articles_data/discovery-search/clip-discovery.png)
+
+Discovery also lets us keep giving feedback to the search engine in the shape of more context pairs, so we can keep refining our search until we find what we are looking for.
+
+Another intuitive example: imagine you're looking for a fish pizza, but pizza names can be confusing, so you can just type "pizza", and prefer a fish over meat. Discovery search will let you use these inputs to suggest a fish pizza... even if it's not called fish pizza!
+
+![Simple discovery example](/articles_data/discovery-search/discovery-example-with-images.png)
+
+## Context search
+
+Now, second case: only providing context.
+
+Ever been caught in the same recommendations on your favourite music streaming service? This may be caused by getting stuck in a similarity bubble. As user input gets more complex, diversity becomes scarse, and it becomes harder to force the system to recommend something different.
+
+![Context vs recommendation search](/articles_data/discovery-search/context-vs-recommendation.png)
+
+__Context search__ solves this by de-focusing the search around a single point. Instead, it selects points randomly from within a zone in the vector space. This search is the most influenced by _triplet loss_, as the score can be thought of as _"how much a point is closer to a negative than a positive vector?"_. If it is closer to the positive one, then its score will be zero, same as any other point within the same zone. But if it is on the negative side, it will be assigned a more and more negative score the further it gets.
+
+![Context search visualization](/articles_data/discovery-search/context-search.png)
+
+Creating complex tastes in a high-dimensional space becomes easier, since you can just add more context pairs to the search. So, in theory, you should be able to constrain the space enough so you select points from a per-search "category" created just from the context in the input.
+
+![A more complex context search](/articles_data/discovery-search/complex-context-search.png)
+
+This way you can give refeshing recommendations, while still being in control by providing positive and negative feedback.
```

Comment:
I found that section quite well explained - was easy to follow for me. however you really got me with the kicker in the front - with the music recommendation. having a real example would be so cool :)